### PR TITLE
Add Purposes functionality and refactor UpdatePersonController

### LINKED ISSUE #55: Add purposes model, asset file, & populate functionality
This includes creating a model, view, update controller, serializer, and URL endpoint, as well as updating the `populate` method so that purposes can be populated from a text file in the `assets` directory.

### PR DESCRIPTION
**Pull request by noah-solomon**
## Overview

Purposes are the reason that a person is using Pear. As of 9/29/2021, these are the possible purposes a user can select:
- Just chatting
- Finding my people
- Meeting someone different
- Learning from mentors
- Guiding mentees
- Not sure yet

These are also available in `/src/assets/pear_purposes.txt` and are populated using `POST /api/populate/` in a similar manner as `Locations`, `Groups`, etc.

In addition to adding the `Purpose` object, `purposes` is now a ManyToMany field in the `Person` model.

I also modified `UpdatePersonController` to reduce duplicated code, and added a helper function to update the ManyToMany sets of a model. This function lives in `/src/api/utils.py`.

## Changes Made

- M src/api/urls.py
- M src/api/utils.py
- M src/api/views.py
- A src/assets/pear_purposes.txt
- M src/pear/settings.py
- M src/person/controllers/update_person_controller.py
- A src/person/migrations/0009_person_purposes.py
- M src/person/models.py
- M src/person/serializers.py
- A src/purpose/__init__.py
- A src/purpose/admin.py
- A src/purpose/apps.py
- A src/purpose/controllers/create_purpose_controller.py
- A src/purpose/controllers/populate_purpose_controller.py
- A src/purpose/controllers/update_purpose_controller.py
- A src/purpose/migrations/0001_initial.py
- A src/purpose/views.py

## Test Coverage

Basic postman testing. TODO: add purposes to the test suite in the shared postman collection

## Next Steps

Add purposes to Notion documentation

## Related PRs or Issues (delete if not applicable)

Closes #55

### CODE REVIEW

File: src/person/controllers/update_person_controller.py [hunk 1]
```diff
@@ -31,6 +33,7 @@ def process(self):
         instagram_username = self._data.get("instagram_username")
         graduation_year = self._data.get("graduation_year")
         pronouns = self._data.get("pronouns")
+        purpose_ids = self._data.get("purposes")
         # goals = self._data.get("goals")
```
Comment (chalo2000) [inline src/person/controllers/update_person_controller.py:37]: might as well get rid of this

File: src/person/serializers.py [hunk 2]
```diff
@@ -41,6 +42,7 @@ class UserSerializer(serializers.ModelSerializer):
     graduation_year = serializers.CharField(source="person.graduation_year")
     pronouns = serializers.CharField(source="person.pronouns")
     # goals = SerializerMethodField("get_goals")
```
Comment (chalo2000) [inline src/person/serializers.py:44]: ditto

File: src/purpose/views.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+    def get(self, request):
+        """Get all purposes."""
+        purposes = Purpose.objects.all()
+        return success_response(
+            self.serializer_class(purposes, many=True).data, status.HTTP_200_OK
+        )
+
+    def post(self, request):
+        """Create an purpose."""
```
Comment (chalo2000) [inline src/purpose/views.py:27]: Create a purpose

File: src/purpose/views.py [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    def post(self, request, id):
+        """Update purpose by id."""
+        try:
+            data = json.loads(request.body)
+        except json.JSONDecodeError:
+            data = request.data
+        return UpdatePurposeController(id, data, self.serializer_class).process()
+
+    def delete(self, request, id):
+        """Delete an purpose by id."""
```
Comment (chalo2000) [inline src/purpose/views.py:57]: a

**Review by chalo2000 — APPROVED** [review]
comment comments but LGTM

File: src/purpose/views.py (hunk 3 above)
Comment (noah-solomon) [reply src/purpose/views.py:27]: haha i always miss these

**Pull request merged**